### DISCUSSION TITLE
Parse Issue (Xcode): Module 'xxxxxxxxx' not found

### DISCUSSION THREAD

**Discussion by printpagestopdf** [codemagic-ci-cd/codemagic-docs#2274] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

My real project fails with this error message that module is not found (e.g Parse Issue (Xcode): Module 'connectivity_plus' not found but happens although for other modules). 
To isolate the problem I cloned the codemagic example [flutter-android-and-ios-yaml-demo-project] and built the ios app successfully without codesigning (by codemagic workflow editor).
In the next step I added the flutter package 'connectivity_plus' to the project. And the build failed with 
Parse Issue (Xcode): Module 'connectivity_plus' not found

Internet search for this error found a good amount of such problems an suggested solutions. None of them solved the problem for this simple project. Beneath other Attempts I tried:
- setting the minimum OS Version to 12.1
- recreating the ios build config with flutter create in the post-clone script (with and without deleting ios/Runner.xcodeproj/project.pbxproj before)
- cleaning pod and pod.cache
- running find . -name "Podfile" -execdir pod install \;
- ......
None of these solved the problem
- 

### Expected results

A Runner.app.zip artefact

### Actual results

Fails with Parse Issue (Xcode): Module 'connectivity_plus' not foun

### Build id (optional)

657df9bd740d53fd79194bb0

**Comment by icarusdust** [comment]
Hi @printpagestopdf, the very first place to check in these cases is whether the package is installed or not.  If you check the logs for `flutter packages pub get`, you should notice that `connectivity_plus` is not getting resolved, which is why the error. 

1. I added `connectivity_plus` in `pubspec.yaml`:
```
dependencies:
  connectivity_plus: ^5.0.2
```
2. Ran `flutter packages pub get`, which gave me the following:
```

Resolving dependencies... 
+ args 2.4.2
> async 2.11.0 (was 2.9.0)
+ connectivity_plus 5.0.2
+ connectivity_plus_platform_interface 1.2.4
```
As you can see the package is resolved unlike yours. 

So what to do => 
 1. First off, check if you are installing the package as explained [here](https://pub.dev/packages/connectivity_plus/install). 
 2. Then you can try `flutter clean` before running ` flutter packages pub get`. You can also try to remove the package from Dart Packages, then run `flutter pub cache repair`. 
 
I strongly suggest you to try all these by remote accessing to the builder machine, so you can try all of these in one go for an easier debugging session. More info [here](https://docs.codemagic.io/troubleshooting/accessing-builder-machine-via-ssh/)

**Reply by printpagestopdf** [reply]
Tried your suggestions, but none of them gave the result that connectivity_plus was listed in flutter packages pub get.

But I got the result this way: removed pubspec.lock from the source and ran flutter packages pub get. Now the depencies 
+ connectivity_plus 5.0.2
+ connectivity_plus_platform_interface 1.2.4
are resolved (beneath others)

BUT build still fails with  "Parse Issue (Xcode): Module 'connectivity_plus' not found"

You can check this in BuildID:  6584815eee610f498d780140

**Comment by printpagestopdf** [comment, marked as answer]
After several trials with different approaches I found this solution for my problem:
Replaced ios/Podfile by this one:
https://fossies.org/linux/flutter/examples/api/ios/Podfile
and the ios app were build without errors!
Don't know how generic this solution is, but perhaps its useful for others.
